builtin/docker/pull/builder.go

```diff
@@ -47,8 +48,8 @@ type BuilderConfig struct {
 	// Control whether or not to inject the entrypoint binary into the resulting image
 	DisableCEB bool `hcl:"disable_entrypoint,optional"`
 
-	// The docker specific encoded authentication string to use to talk to the registry.
-	EncodedAuth string `hcl:"encoded_auth,optional"`
```

Comment:
So we can't delete this because we'll break everyone that is using it, so the new Auth mechanism has to work along side EncodedAuth (probably just use EncodedAuth if set, otherwise the new Auth).

Reply:
I'll work that back in there for backwards-compatibility.

Reply:
Still working on the registry piece, but for the puller, all of these test cases now work as expected:

```hcl
    //auth used for private repo, expected build success
    workspace "pull" {
      use "docker-pull" {
        image = "paladindevops/alpine"
        tag   = "very-latest"
        auth {
          username = "username"
          password = "password"
        }
      }
    }

    //encoded_auth used for private repo, expected build success
    workspace "pull-encoded" {
      use "docker-pull" {
        image = "paladindevops/alpine"
        tag = "very-latest"
        encoded_auth = "encoded_docker_auth"
      }
    }

    //no auth used for private repo, expected build failure
    workspace "pull-noauth" {
      use "docker-pull" {
        image = "paladindevops/alpine"
        tag = "very-latest"
      }
    }

    //no auth used for public repo, expected build success
    workspace "pull-public" {
      use "docker-pull" {
        image = "alpine"
        tag = "latest"
      }
    }
```